app/components/documents/document-details-panel.js

```diff
@@ -97,6 +97,14 @@ export default class DocumentsDocumentDetailsPanel extends Component {
 
   @task
   *saveDetails() {
+    // This is to avoid concurrency issues where the piece
+    // might have changed already
+    try {
+      const oldPiece = yield this.args.piece;
+      yield oldPiece.save();
+    } catch (e) {
+      return;
+    }
```

Comment:
why save here? I'm guessing to test concurrency?
if you are going to use `ModelWithModifier` you can just use `preEditOrSaveCheck()` to check if anyone else made changes to piece before you start the process below.